src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,83 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    void register_addingNewUser_Ok() {
+        User userOk = new User();
+        userOk.setLogin("NewUser");
+        userOk.setPassword("123456789");
+        userOk.setAge(18);
+        User actual = registrationService.register(userOk);
+        assertEquals(userOk, actual);
+    }
+
+    @Test
+    void register_addingUserWhichAlreadyExist_NotOk() {
+        User userDouble = new User();
+        userDouble.setLogin("userDouble");
+        userDouble.setPassword("123456789");
+        userDouble.setAge(18);
+        registrationService.register(userDouble);
+        User actual = registrationService.register(userDouble);
+        assertNull(actual);

```

Comment:
The test `register_addingUserWhichAlreadyExist_NotOk` expects `null` when trying to register a user that already exists. However, the `RegistrationServiceImpl` should ideally throw an exception or handle this case differently, as returning `null` is not informative. Consider updating the implementation and the test to reflect a more appropriate behavior, such as throwing an `InvalidDataException`.